docs/widgets/app_launcher.md

```diff
@@ -0,0 +1,126 @@
+    search_commands = true                                            -- Search by app name AND commandline command
+    skip_names = { "Discord" }                                        -- List of apps to omit from launcher
+    skip_commands = { "thunar" }                                      -- List of commandline commands to omit from launcher
+    skip_empty_icons = true                                           -- Skip applications without icons
+    sort_alphabetically = true                                        -- Sorts applications alphabetically
+    reverse_sort_alphabetically = false                               -- Sort in reverse alphabetical order (NOTE: must set `sort_alphabetically = false` to take effect)
+    select_before_spawn = true                                        -- When selecting by mouse, click once to select app, click once more to open the app.
+    hide_on_left_clicked_outside = true                               -- Hide launcher on left click outside the launcher popup
+    hide_on_right_clicked_outside = true                              -- Hide launcher on right click outside the launcher popup
+    hide_on_launch = true                                             -- Hide launcher when spawning application
+    try_to_keep_index_after_searching = false                         -- After a search, reselect the previously selected app
+    reset_on_hide = true                                              -- When you hide the launcher, reset search query
+    save_history = true                                               -- Save search history
+    wrap_page_scrolling = true                                        -- Allow scrolling to wrap back to beginning/end of launcher list
+    wrap_app_scrolling = true                                         -- Set app scrolling 
+
+    default_app_icon_name = "standard.svg"                            -- Sets default app icon name for apps without icon names
+    default_app_icon_path = "~/icons/"                                -- Sets default app icon path for apps without icon paths
+    icon_theme = "application"                                        -- Set icon theme
+    icon_size = 24                                                    -- Set icon size
+
+    type = "dock"                                                     -- awful.popup type ("dock", "desktop", "normal"...).  See awesomewm docs for more detail
+    show_on_focused_screen = true                                     -- Should app launcher show on currently focused screen
+    screen = awful.screen                                             -- Screen you want the launcher to launch to
+    placement = awful.placement.top_left                              -- Where launcher should be placed ("awful.placement.centered").
+    rubato = { x = rubato_animation_x, y = rubato_animation_y }       -- Rubato animation to apply to launcher
```

Comment:
I'll have to defer to you on this rubato option.  I could not get anything to work in my testing, but I have never used rubato before and I don't know when I'm going to have the time/energy to go down that rabbit hole.